integrations/openai/src/ucai_openai/toolkit.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Dict, List, Optional
+
+from openai import pydantic_function_tool
+from openai.types.chat import ChatCompletionToolParam
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+from unitycatalog.ai.client import BaseFunctionClient
+from unitycatalog.ai.utils.client_utils import validate_or_set_default_client
+from unitycatalog.ai.utils.function_processing_utils import (
+    generate_function_input_params_schema,
+    get_tool_name,
+    process_function_names,
+)
+
+
+class UCFunctionToolkit(BaseModel):
+    function_names: List[str] = Field(
+        default_factory=list,
+        description="The list of function names in the form of 'catalog.schema.function'",
+    )
+
+    client: Optional[BaseFunctionClient] = Field(
+        default=None,
+        description="The client for managing functions, must be an instance of BaseFunctionClient",
+    )
+
+    tools_dict: Dict[str, ChatCompletionToolParam] = Field(
+        default_factory=dict,
+        description="The tools dictionary storing the function name and tool definition mapping, no need to provide this field",
+    )
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @model_validator(mode="after")
+    def validate_toolkit(self) -> "UCFunctionToolkit":
+        self.client = validate_or_set_default_client(self.client)
+
+        self.tools_dict = process_function_names(
+            function_names=self.function_names,
+            tools_dict={},
+            client=self.client,
+            uc_function_to_tool_func=self.uc_function_to_openai_function_definition,
+        )
+        return self
+
+    @classmethod
```

Comment:
we can use a static method. I'd use a class method for a factory such as `from_dict`.

Reply:
Make sense